nerfstudio/field_components/encodings.py

```diff
@@ -381,6 +426,31 @@ def pytorch_fwd(self, in_tensor: Float[Tensor, "*bs input_dim"]) -> Float[Tensor
 
         return torch.flatten(encoded_value, start_dim=-2, end_dim=-1)  # [..., num_levels * features_per_level]
 
+    def regularize_hash_pyramid(
+        self,
+        regularize_fn: Callable[[Tensor], Tensor] = torch.abs,
+    ) -> Float[Tensor, "0"]:
+        """Regularize hash pyramid weights."""
+        hash_decay = segment_coo(
```

Comment:
For avoiding torch_scatter dependency
```suggestion
        hash_decay = torch.zeros((self.num_levels, 2), device=self.hash_table.device)
        hash_decay = hash_decay.scatter_reduce(
            dim=0,
            index=self.level_indexes[..., None].repeat(1, self.features_per_level),
            src=regularize_fn(self.hash_table.view(-1, self.features_per_level)),
            reduce="mean",
        ).mean(dim=-1).sum()
```

Reply:
Thanks! I kept in mind this solution. but how to solve dependency with `scale_featurization` ?

Reply:
I suppose
```
scale_feat = torch.zeros(self.num_levels, device=self.hash_table.device)
scale_feat = scale_feat.scatter_reduce(
            dim=0,
            index=self.level_indexes,
            src=self.hash_table.view(-1, self.features_per_level).pow(2).sum(-1),
            reduce="mean",
)
```
should do the trick?

Reply:
thanks! this works!

Reply:
@georghess 
<img width="974" alt="Screenshot 2023-10-22 at 19 14 28" src="https://github.com/nerfstudio-project/nerfstudio/assets/25929618/38abac6c-bdb0-4db1-92c3-4f0e1d49e1df">

> For avoiding torch_scatter dependency

Reply:
Sorry, 
`hash_decay = torch.zeros((self.num_levels, 2), device=self.hash_table.device)`
should be
`hash_decay = torch.zeros((self.num_levels, self.features_per_level), device=self.hash_table.device)`

However, for my applications, a simple for loop was way faster.